src/sentry/grouping/strategies/hierarchical.py

```diff
@@ -27,7 +27,7 @@ def get_stacktrace_hierarchy(
         assert key not in all_variants
 
         found_sentinel = False
-
+        component = None
```

Comment:
Just for my understanding, is this necessary? Or would the `if not found_sentinel` on line 52 break out of the outer loop anyway if component is not set?

Reply:
So this is because I got lint screaming at me 
```shell
sentry/grouping/strategies/hierarchical.py:55:25 - error: "component" is unbound (reportUnboundVariable)
sentry/grouping/strategies/hierarchical.py:57:26 - error: "component" is unbound (reportUnboundVariable)
```
Maybe you have a better idea how to make it pass 😄 

The rest of the code works as expected, yes, and basically without any change

Reply:
All good!